tests/v2/validation/rbac/rbac_psa_test.go

```diff
@@ -0,0 +1,336 @@
+		require.NoError(rb.T(), err)
+	case roleMember, roleProjectOwner, roleProjectMember, roleProjectReadOnly:
+		require.Error(rb.T(), err)
+		if role == roleMember || role == roleProjectReadOnly {
+			errMessage := strings.Split(err.Error(), ":")[0]
+			assert.Equal(rb.T(), "Resource type [namespace] is not creatable", errMessage)
+		} else {
+			errStatus := strings.Split(err.Error(), ".")[1]
+			rgx := regexp.MustCompile(`\[(.*?)\]`)
+			errorMsg := rgx.FindStringSubmatch(errStatus)
+			assert.Equal(rb.T(), "403 Forbidden", errorMsg[1])
+		}
+	}
+
+}
+
+func (rb *PSATestSuite) ValidateAdditionalPSA(role string) {
+	createProjectAsNonAdmin, err := createProject(rb.nonAdminUserClient, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+
+	namespaceName := namegen.AppendRandomString("testns-")
+	createNamespace, err := namespaces.CreateNamespace(rb.nonAdminUserClient, namespaceName, "{}",
+		map[string]string{}, map[string]string{},
+		createProjectAsNonAdmin)
+	require.NoError(rb.T(), err)
+
+	rb.T().Logf("Validate editing new namespace in a cluster member created project with PSA labels as %v", role)
+	labels := map[string]string{
+		psaWarn:    pssRestrictedPolicy,
+		psaEnforce: pssRestrictedPolicy,
+		psaAudit:   pssRestrictedPolicy,
+	}
+
+	updateNS := rb.ConvertNS(createNamespace)
+	updateNS.Labels = labels
+
+	relogin, err := rb.nonAdminUserClient.ReLogin()
+	require.NoError(rb.T(), err)
+	rb.nonAdminUserClient = relogin
+
+	steveStdUserclient, err := rb.nonAdminUserClient.Steve.ProxyDownstream(rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	rb.steveNonAdminClient = steveStdUserclient
+
+	response, err := rb.steveNonAdminClient.SteveType(namespaces.NamespaceSteveType).Update(createNamespace, updateNS)
+
+	switch role {
+	case roleOwner:
+		require.NoError(rb.T(), err)
+		expectedLabels := getPSALabels(response, labels)
+		assert.Equal(rb.T(), labels, expectedLabels)
+		_, err = createDeployAndWait(rb.steveNonAdminClient, rb.client, rb.cluster.ID, "nginx", "nginx", createNamespace.Name)
+		require.Error(rb.T(), err)
+	case roleMember:
+		require.Error(rb.T(), err)
+		errStatus := strings.Split(err.Error(), ".")[1]
+		rgx := regexp.MustCompile(`\[(.*?)\]`)
+		errorMsg := rgx.FindStringSubmatch(errStatus)
+		assert.Equal(rb.T(), "403 Forbidden", errorMsg[1])
+
+		//Edit the namespace and add PSA labels as an admin for a cluster member
+		updateNS := rb.ConvertNS(createNamespace)
+		updateNS.Labels = labels
+		_, err := rb.steveAdminClient.SteveType(namespaces.NamespaceSteveType).Update(createNamespace, updateNS)
+		require.NoError(rb.T(), err)
+
+	}
+
+	rb.T().Logf("Validate deletion of PSA labels in namespace in a cluster member created project as %v", role)
+
+	for label := range labels {
+		if strings.Contains(label, psaWarn) || strings.Contains(label, psaAudit) || strings.Contains(label, psaEnforce) {
+			delete(labels, label)
+		}
+	}
```

Comment:
I think as Israel mentioned, it seems like it's going to be used multiple times, so I think it might be beneficial to make this a helper in `rbac.go`